renter/hostdb/weight.go

```diff
@@ -0,0 +1,465 @@
+package hostdb
+
+import (
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.sia.tech/core/types"
+)
+
+// An Allowance dictates how much the Renter is allowed to spend in a given
+// period. Note that funds are spent on both storage and bandwidth.
+//
+// NOTE: When changing the allowance struct, any new or adjusted fields are
+// going to be loaded as blank when the contractor first starts up. The startup
+// code either needs to set sane defaults, or the code which depends on the
+// values needs to appropriately handle the values being empty.
+type Allowance struct {
+	Funds       types.Currency `json:"funds"`
+	Hosts       uint64         `json:"hosts"`
+	Period      uint64         `json:"period"`
+	RenewWindow uint64         `json:"renewwindow"`
+
+	// ExpectedStorage is the amount of data that we expect to have in a contract.
+	ExpectedStorage uint64 `json:"expectedstorage"`
+
+	// ExpectedUpload is the expected amount of data uploaded through the API,
+	// before redundancy, per block.
+	ExpectedUpload uint64 `json:"expectedupload"`
+
+	// ExpectedDownload is the expected amount of data downloaded through the
+	// API per block.
+	ExpectedDownload uint64 `json:"expecteddownload"`
+
+	// ExpectedRedundancy is the average redundancy of files being uploaded.
+	ExpectedRedundancy float64 `json:"expectedredundancy"`
+
+	// MaxPeriodChurn is maximum amount of contract churn allowed in a single
+	// period.
+	MaxPeriodChurn uint64 `json:"maxperiodchurn"`
+}
+
+// acceptContractAdjustments checks that a host which doesn't accept contracts
+// will receive the worst score possible until it enables accepting contracts
+// again.
+func (db *DB) acceptContractAdjustments(host Host) float64 {
+	if !host.Settings.AcceptingContracts {
+		return math.SmallestNonzeroFloat64
+	}
+	return 1
+}
+
+// lifetimeAdjustments will adjust the weight of the host according to the total
+// amount of time that has passed since the host's original announcement.
+func (db *DB) lifetimeAdjustments(host Host) float64 {
+	index, err := db.store.Index()
+	if err != nil {
+		return math.SmallestNonzeroFloat64
+	}
```

Comment:
better to retrieve this value in `Score`, and pass it to `lifetimeAdjustments` as a parameter.

As a general note, I suggest making these functions standalone, rather than methods on `db`.